src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -40,14 +40,27 @@ import LogOutPreviousUserPage from '../../../pages/LogOutPreviousUserPage';
 import networkPropTypes from '../../../components/networkPropTypes';
 import {withNetwork} from '../../../components/OnyxProvider';
 
+let currentUserEmail;
 Onyx.connect({
-    key: ONYXKEYS.MY_PERSONAL_DETAILS,
+    key: ONYXKEYS.SESSION,
     callback: (val) => {
+        // When signed out, val is undefined
         if (!val) {
             return;
         }
 
-        const timezone = lodashGet(val, 'timezone', {});
+        currentUserEmail = val.email;
+    },
+});
+
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS,
+    callback: (val) => {
+        if (!val) {
+            return;
+        }
+
+        const timezone = lodashGet(val, currentUserEmail, 'timezone', {});
```

Comment:
NAB, it seems like there is a very remote chance that `currentUserEmail` won't have a value at this point. It depends entirely on the `session.email` data being set before the `personalDetails` and both would be set when the user logs in. 

The subscriptions here feel out of place though and should be probably be moved into `PersonalDetails` (but not necessary to do that now).